### PR TITLE
Add test module for ZNC and connect using weechat

### PR DESCRIPTION
**Pull request by asdil12**
This will help us catching bugs like https://bugzilla.suse.com/show_bug.cgi?id=1108450 in the future.

Verification: http://artemis.suse.de/tests/536#step/weechat/8
Needles: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/443

### CODE REVIEW

File: tests/console/znc.pm [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
```
Comment (okurz) [inline tests/console/znc.pm]: Would you mind using the BSD type license instead we also use in test modules?

File: tests/console/znc.pm [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+# Summary: Setup ZNC IRC bouncer
```
Comment (okurz) [inline tests/console/znc.pm]: Can you better describe here what is the testing goal and not what it does?

File: tests/console/weechat.pm [hunk 3]
```diff
@@ -27,6 +27,17 @@ sub run {
...
     assert_screen('weechat');
+
+    type_string("/server add znc localhost/12345 -ssl -ssl_verify=0 -username=bernhard/freenode -password=$testapi::password\n");
+    assert_screen('weechat-server-added');
+
+    type_string("/connect znc\n");
+    assert_screen('weechat-no-longer-away');
+
+    type_string("/query *status\n");
+    type_string("Version\n");
```
Comment (okurz) [inline tests/console/weechat.pm]: That could be a bit tricky as you want that command to show up in the query. Is it ensured to be open when you type this

File: tests/console/znc.pm [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+use strict;
+use testapi;
+use utils qw(zypper_call systemctl);
+
+sub run {
+    select_console('root-console');
+    zypper_call('in znc');
+
+    script_run("sudo -u znc znc --makeconf; echo zncconf-status-\$? > /dev/$serialdev", 0);
+    sleep 1;
```
Comment (okurz) [inline tests/console/znc.pm]: Please get rid of all sleeps and replace by other sync methods. We have good examples in other test modules

File: tests/console/weechat.pm (hunk 3 above)
Comment (asdil12) [reply tests/console/weechat.pm]: In my tests it never missed the query.
IIRC the query command doesn't require any server interaction so it should be quite fast.

File: tests/console/znc.pm (hunk 4 above)
Comment (asdil12) [reply tests/console/znc.pm]: removed the sleeps

File: tests/console/znc.pm (hunk 1 above)
Comment (asdil12) [reply tests/console/znc.pm]: replaced license with something from another module

**Comment by okurz** [comment]
LGTM. I saw you verified on Tumbleweed. Please test on Leap as well or merge yourself and ensure to handle the failures in production

**Comment by asdil12** [comment]
Verification for leap: http://artemis.suse.de/tests/541

**Pull request merged**